Add periodic socok8s airship from rpm job

jhesketh opened the pull request:
This tests the socok8s project from RPM rather than from source. As we
are testing an RPM, rather than installing it onto the host worker,
install into and operate it from a podman container.

This job requires the jenkins worker to be set up with podman. This has
already been done manually for the cloud-ccp-ci node.

toabctl commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,184 @@
+            }
+        }
+
+        stage('Create network') {
```
Is it possible to move theses stages to a library or something? So we don't need to copy the steps into multiple Jenkinsfile's?

jhesketh replied:
Maybe, but because the jobs are in different repositories it becomes more complex. If we move the socok8s PR jenkins integration into this automation repo, it might be easier.

Given the simplicity of these steps I would be hesitant to add the complexity needed to make them a library. The real complexity inside those steps is already shared by being part of socok8s.

it is, however, a possible future improvement we could look at.

toabctl replied:
Ah. True. I would move the Jenkinsfile to the socok8s repo. Otherwise, we can not test changes to that file with the current PR testing. And then we might be able to share stages between the jobs.

jhesketh replied:
The difference is that this isn't testing what is in git - this only tests what is in OBS. So it didn't make as much sense to put in the socok8s repo IMO. Granted, if we do look to turn these into a library it may make sense then.

toabctl replied:
We test at least the Jenkinsfile itself if we move it to the socok8s git repo

jhesketh replied:
Yes, but these are (in my opinion) becoming different enough to lose any benefit of trying to turn them into a library. There is a bunch of extra logic going into the socok8s integration job to determine when to run which step. This job always knows what steps it wants to run and is a subset of the available ones.

You can test this file by triggering a manual run of the job in jenkins and setting the repo+branch of where to pull it from. That is how I have been running the tests on the aforementioned example runs.

toabctl replied:
ok. I still don't like the amount of code we duplicate here but ok for now I guess.

jsuchome commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,184 @@
+                    //                  (details and alternative fix below)
+                    in_container('zypper addrepo http://download.opensuse.org/repositories/Cloud:/OpenStack:/Rocky/openSUSE_Leap_15.0/ "Cloud OpenStack Rocky"')
+                    // Install CA and socok8s
+                    in_container('zypper addrepo http://download.suse.de/ibs/SUSE:/CA/openSUSE_Leap_15.0/ "SUSE CA"')
```
Why do we need CA repo?

JanZerebecki replied:
To access the cloud where this creates its heat stacks.

jhesketh commented:
Rebased, and removed "teardown" question.

Successful example run here: https://ci.suse.de/blue/organizations/jenkins/jhesketh-socok8s-testing-project/detail/jhesketh-socok8s-testing-project/57/pipeline

toabctl commented on jenkins/ci.suse.de/cloud-socok8s.yaml:
```diff
@@ -6,3 +6,4 @@
     repo-credentials: c2350527-476a-45df-b406-84f028614682
     jobs:
       - '{name}-integration'
+      - '{name}-nightly-airship-from-rpm'
```
Now that we have `stable/1.0` and `master`, I think we need 2 jobs

jhesketh replied:
This requires a bit of reworking. Can we do that in a followup?

toabctl replied:
Yes. True

jsuchome commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,181 @@
+                    // zypper --gpg-auto-import-keys refresh
+                    // zypper install -yl podman
+                    // usermod --add-subuids 65536-362144 jenkins
+                    // usermod --add-subgids 65536-362144 jenkins
```
Why is this block commented?

jhesketh replied:
As per the first line, this is what is required to set up the jenkins worker as a once off. It is left here as documentation on how to set up a new node if the following `which podman` were to fail.

jsuchome replied:
Thanks. Could you please update the comment so it is clearer? `(already applied)` did not make it clear for me

bmwiedemann replied:
Could it also be wrapped in some `if initial_jenkins_worker_setup` block?

jhesketh replied:
That would require the jenkins user to become root. I just applied these manually. If there is a better place to document the worker setup, please let me know.

JanZerebecki replied:
Some jenkins workers are set up with https://gitlab.suse.de/cloud/cloudsalt/ however see https://jira.suse.com/browse/SOC-9376 .

jhesketh replied:
Yes, ideally (in the future) this should go into some config management for the node. Our current CCP test node is only set up by hand though.

jsuchome commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,181 @@
+                    // FIXME(jhesketh): the openstacksdk library MUST be >=0.12.0 for ansible.
+                    //                  Install from pip for now until we resolve the proper deps
+                    //in_container('zypper --no-gpg-checks install -yl python3-pip')
+                    //in_container('pip3 install -U openstacksdk==0.27.0')
```
So why is it commented if we want to do it?

jhesketh replied:
The alternate fix above of adding the Cloud:Rocky repository fixes this. This is largely left over from my debugging and once https://bugzilla.suse.com/show_bug.cgi?id=1137590 closes can be removed.

jsuchome replied:
I see. Again, if you want to keep these kind of comments, please write a note directly there why is it commented

jhesketh replied:
Done

toabctl commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,181 @@
+                    in_container('zypper addrepo http://download.opensuse.org/repositories/Cloud:/OpenStack:/Rocky/openSUSE_Leap_15.0/ "Cloud OpenStack Rocky"')
+                    // Install CA and socok8s
+                    in_container('zypper addrepo http://download.suse.de/ibs/SUSE:/CA/openSUSE_Leap_15.0/ "SUSE CA"')
+                    in_container('zypper addrepo https://download.opensuse.org/repositories/Cloud:/socok8s/openSUSE_Leap_15.0/ "SUSE OpenStack Cloud on Kubernetes Preview"')
```
This should be parameterized so we can add `Cloud:/socok8s` and `Cloud:/master:/socok8s`.
And if possible, having the distro as parameter would be also good. I think we'll move soon to Leap15.1

Itxaka replied:
Agreed but should we tackle this in a later patch?

toabctl commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,181 @@
+                    in_container('zypper --no-gpg-checks refresh')
+                    in_container('zypper --no-gpg-checks install -yl ca-certificates-suse socok8s')
+
+                    // FIXME(jhesketh): the openstacksdk library MUST be >=0.12.0 for ansible.
```
Could you mention the opensuse bug number so we know why we need this workaround?

jhesketh replied:
Done

Itxaka commented on jenkins/ci.suse.de/pipelines/socok8s-airship-from-rpm.Jenkinsfile:
```diff
@@ -0,0 +1,181 @@
+            zip archive: true, dir: 'logs/', zipFile: 'logs.zip'
+            archiveArtifacts artifacts: 'logs.zip'
+        }
+        cleanup {
```
should we bring the gather_logs part from the PR job in here as well?

jhesketh replied:
Sorry, could you please link me to which part you mean?

Itxaka replied:
https://github.com/SUSE-Cloud/socok8s/blob/master/Jenkinsfile.integration#L189

Itxaka replied:
this can come later thougth

jhesketh replied:
This is already there in the failure part just above.
